### PR TITLE
Added hooktube embed functionality

### PR DESCRIPTION
**Pull request by Chiiruno**
I can't get the meguca binary to work (even without my changes), so I can't really test it, but theoretically this should work. At least the /embed one should work on body_test.go. It compiles, at least.

### CODE REVIEW

**Comment by Chiiruno** [comment]
Figured out why the meguca binary was having that getopt issue. Had nothing to do with my local connection or port issues, but rather with postgresql. meguca is doing the systemwide db instead of a local db, so I have to create the db piece by piece, which is annoying. make postgresql user -> make db -> column "locked" of relation "threads" does not exist -> etc...

**Comment by bakape** [comment]
>a local db
What do you mean by this?

**Comment by Chiiruno** [comment]
> What do you mean by this?
Who knows. I haven't messed with websites or databases in 8 years.

File: client/posts/embed.ts [hunk 1]
```diff
@@ -35,7 +43,7 @@ const formatters: { [key: number]: (s: string) => string } = {}
 // Map of providers to information fetcher functions
 const fetchers: { [key: number]: (el: Element) => Promise<void> } = {}
 
-for (let p of ["Youtube", "SoundCloud", "Vimeo"]) {
+for (let p of ["Youtube", "Hooktube", "SoundCloud", "Vimeo"]) {
```
Comment (bakape) [inline client/posts/embed.ts]: These are called through `noembed.com`. You need a separate handler (or 2?) for hooktube. Some sort of error handling would be nice too.

File: client/posts/embed.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ type OEmbedDoc = {
 }
 
 // Types of different embeds by provider
-enum provider { Youtube, SoundCloud, Vimeo }
+enum provider { Youtube, Hooktube, SoundCloud, Vimeo }
```
Comment (bakape) [inline client/posts/embed.ts]: Could you append to the enum instead to break running applications less?

File: go/src/meguca/templates/body.go [hunk 3]
```diff
@@ -18,6 +18,7 @@ import (
 // Embeddable URL types
 const (
 	youTube = iota
+	hookTube
```
Comment (bakape) [inline go/src/meguca/templates/body.go]: Same with the enum here.

File: go/src/meguca/templates/body_test.go [hunk 4]
```diff
@@ -279,6 +279,16 @@ func TestRenderBody(t *testing.T) {
 			out:  "<em><a rel=\"noreferrer\" class=\"embed\" target=\"_blank\" data-type=\"0\" href=\"https://youtu.be/z0f4Wgi94eo\">[Youtube] ???</a></em>",
 		},
 		{
+			name: "hooktube embed",
+			in:   "https://www.hooktube.com/watch?v=z0f4Wgi94eo",
+			out:  "<em><a rel=\"noreferrer\" class=\"embed\" target=\"_blank\" data-type=\"0\" href=\"https://www.hooktube.com/watch?v=z0f4Wgi94eo\">[Hooktube] ???</a></em>",
```
Comment (bakape) [inline go/src/meguca/templates/body_test.go]: `data-type` value must match enum.

**Comment by bakape** [comment]
If you still have trouble with getting the server to run, feel free to ask.

File: client/posts/embed.ts (hunk 2 above)
Comment (Chiiruno) [reply client/posts/embed.ts]: Now that I look at that, you're right. The datatype stuff on the bottom makes sense too now.
Sorry about that, I'll fix it.

**Comment by Chiiruno** [comment]
Yeah, I'd like to know how to generate the database.
Other than that, I fixed the issues. Not sure what I can do about a separate handler though, I'm very new to Go and have never dabbled in Javascript.

**Comment by bakape** [comment]
The instructions in `docs/installation.md` should be clear enough. Let me check this then.

File: client/posts/embed.ts [hunk 5]
```diff
@@ -35,7 +43,7 @@ const formatters: { [key: number]: (s: string) => string } = {}
 // Map of providers to information fetcher functions
 const fetchers: { [key: number]: (el: Element) => Promise<void> } = {}
 
-for (let p of ["Youtube", "SoundCloud", "Vimeo"]) {
+for (let p of ["Youtube", "SoundCloud", "Vimeo", "Hooktube"]) {
```
Comment (bakape) [inline client/posts/embed.ts]: This still calls into the noembed.com API (`formatNoEmbed()` and `fetchNoEmbed()`). You will need figure out how to handle the hooktube API and write separate functions for that.

**Comment by Chiiruno** [comment]
I followed installation.md, still get stuck at `pq: column "locked" of relation "threads" does not exist` after initializing the database.

the data is all gone

**Comment by bakape** [comment]
>I followed installation.md, still get stuck at pq: column "locked" of relation "threads" does not exist after initializing the database.

Yup, bug. Thanks for finding it. Working on a fix.

**Comment by bakape** [comment]
@Chiiruno 
Fixed. Just pull the latest changes and it should work.

**Comment by Chiiruno** [comment]
Worked like a charm, thanks for the fix.
As far as hooktube goes, the `make test` error should go away whenever leedo accepts the noembed pull, and hooktube embedding should work. Do you want to keep the pull request up for a while, or should I comment out the changes until said noembed pull is ready?
That said, if leedo doesn't respond within a month, I'll just write a new method to handle hooktube links, but it's not the ideal way to handle this.

**Comment by bakape** [comment]
Yeah, I think it's best to just keep this open till then and wait for a response. No need to commit, in case he never merges the change.

**Comment by bakape** [comment]
@Chiiruno 
Any updates on this?

**Comment by Chiiruno** [comment]
@bakape Ah. I completely forgot about this.
Sorry, I kind of lost interest. I'll take down the pull request since that leedo faggot isn't responding to anything.

**Pull request closed without merging**